jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -16,37 +16,92 @@
 
 package com.google.cloud.tools.jib.image;
 
-import com.google.common.collect.ImmutableList;
+import com.google.common.annotations.VisibleForTesting;
 import java.nio.file.Path;
+import java.util.Objects;
 
 /**
  * Represents an entry in the layer. A layer consists of many entries that can be converted into tar
  * archive entries.
  */
 public class LayerEntry {
 
+  private final Path sourceFile;
+  private final Path extractionPath;
+
   /**
-   * The source files to build from. Source files that are directories will have all subfiles in the
-   * directory added (but not the directory itself).
+   * Instantiates with a source file and the path to place the source file in the container file
+   * system.
    *
-   * <p>The source files are specified as a list instead of a set to define the order in which they
-   * are added.
+   * @param sourceFile the source file to add to the layer
+   * @param extractionPath the path to place the source file in the container file system (relative
+   *     to root {@code /})
    */
-  private final ImmutableList<Path> sourceFiles;
-
-  /** The Unix-style path to add the source files to in the container image filesystem. */
-  private final String extractionPath;
-
-  public LayerEntry(ImmutableList<Path> sourceFiles, String extractionPath) {
-    this.sourceFiles = sourceFiles;
+  public LayerEntry(Path sourceFile, Path extractionPath) {
+    this.sourceFile = sourceFile;
     this.extractionPath = extractionPath;
   }
 
-  public ImmutableList<Path> getSourceFiles() {
-    return sourceFiles;
+  /**
+   * Gets the source file. Do <b>not</b> call {@link Path#toString} on this - use {@link
```

Comment:
Curious. Will it be wrong to call `toString()`? Can `getSourceFile().toString()` and `getSourceFileString()` be different? The emphasis on "not" gives me the impression that they can be different.